Feature/reg 1252

urquha opened the pull request:
Added zipkin tracing following the sbr-control-api implementation

Commit:
tracing added

Commit:
typo

Commit:
typo

Commit:
Variable change

Commit:
added readme

nigelhp commented on conf/application.conf:
```diff
@@ -234,3 +234,24 @@ hbase {
+
+kamon {
+  environment {
+    service = "sbr-admin-data"
```
For admin data, it may be a good idea to allow the service name to be overriden by an environment variable (using the usual pattern where the default value is followed by another declaration specifying ```service = ${?ENVIRONMENT_VARIABLE_NAME})```.

There will be 3 different instances of the admin data service (paye, vat, ch) - so it might be nice if they can be configured to have a service name of sbr-admin-data-vat for example.

nigelhp replied:
Note that to achieve this the Jenkins build will need to change, as each variable needs to be explicitly injected into the deployment (given the way it is currently setup).

nigelhp commented on conf/application.conf:
```diff
@@ -234,3 +234,24 @@ hbase {
+    port = 9411
+    port = ${?SBR_TRACING_REPORTER_PORT}
+  }
+}
```
In order to consider this story "done done", we should make sure that the correct Zipkin configuration is added to the Gitlab manifest files.

(Note that the Gitlab project used is confusingly the one for the old name - i.e the one ending -api).

nigelhp commented on build.sbt:
```diff
@@ -84,6 +87,9 @@ lazy val `sbr-admin-data` = (project in file("."))
     routesImport += "extensions.Binders._",
       //moduleName := "sbr-admin-data",
     description := "<description>",
+    javaOptions in Test += "-DSBR_DB_PORT=8075",
```
I don't think we need this test port specification

nigelhp reviewed (dismissed):
Please see comments relating to some things we should do to consider this fully "done done" and ready for deployment.

Commit:
some pull request comments addressed

Commit:
Merge branch 'master' into feature/REG-1252

Commit:
Merge branch 'master' into feature/REG-1252

Commit:
Rename tracing app name env variable

- Renamed tracing app name env variable to a more appropriate name of
SBR_TRACING_APP_NAME

Commit:
Merge branch 'master' into feature/REG-1252
